test/Concurrency/async_initializer.swift

```diff
@@ -0,0 +1,138 @@
+// RUN: %target-typecheck-verify-swift -enable-experimental-concurrency
+// REQUIRES: concurrency
+
+
+////
+// some functions to play with
+func f() async {
+  let _ = Person() // expected-error {{call is 'async' but is not marked with 'await'}} {{11-11=await }}
+}
+
+func g() {}
+
+////
+// test super.init interactions
+
+class Person {
+  init() async {
+    f() // expected-error {{call is 'async' but is not marked with 'await'}} {{5-5=await }}
+  }
+
+  convenience init(_ s: String) async {
+    await self.init()
+  }
+}
+
+class Bertrand: Person {
+  override init() {} // expected-error {{missing call to superclass's initializer; 'super.init' is 'async' and requires an explicit call}}
+  init(_ x: Int) async {} // expected-error {{missing call to superclass's initializer; 'super.init' is 'async' and requires an explicit call}}
+}
+
+class Barbara: Person {
+  // expected-note@+2 {{add 'async' to function 'init(_:)' to make it asynchronous}} {{20-20= async}}
+  // expected-error@+1 {{missing call to superclass's initializer; 'super.init' is 'async' and requires an explicit call}}
+  init(_ d: Double) {
+    f() // expected-error{{'async' call in a function that does not support concurrency}}
+  }
+
+  init(x: Int, y: Int) async {
+    await super.init()
+  }
+
+  convenience init(a: Double, b: Double) async {
+    await self.init(x: 0, y: 0)
+  }
+}
+
+class Fruit {
+  init() async {}
+  init(name: String) {}
+}
+
+class Banana: Fruit {
+  override init() {
+    super.init(name: "banana")
+  }
+}
+
+class Cat {} // expected-note {{overridden declaration is here}}
+
+class Calico: Cat {
+  override init() async {} // expected-error {{cannot override non-async initializer with async initializer}}
+}
+
+func reconstruct(c: Cat) {
+  c.init() // expected-error {{'init' is a member of the type}}
+}
+
+////
+// test reasync initializers
+
+class MyType {
+  init(_ f: () async -> Void) reasync {
+    await f()
+  }
+}
+
+func beep() async {
+  let _ = MyType(f) // expected-error{{call is 'async' but is not marked with 'await'}}
+  let _ = await MyType(f)
+
+  let _ = MyType(g)
+}
+
+////
+// test other types with constructors
+
+actor A {
```

Comment:
Could you also add some examples with global actors and async initializers?

Reply:
👍🏼 done. thanks for pointing this out